Applications/SlicerApp/Testing/Python/DICOMReaders.py

```diff
@@ -0,0 +1,189 @@
+import logging
+import os
+import unittest
+import urllib
+import vtk, qt, ctk, slicer
+from slicer.ScriptedLoadableModule import *
+from DICOMLib import DICOMUtils
+
+#
+# DICOMReaders
+#
+
+class DICOMReaders(ScriptedLoadableModule):
+  def __init__(self, parent):
+    ScriptedLoadableModule.__init__(self, parent)
+    parent.title = "DICOMReaders"
+    parent.categories = ["DICOMReaders"]
+    parent.dependencies = []
+    parent.contributors = ["Steve Pieper (Isomics)"]
+    parent.helpText = """
+    This module was developed to confirm that different DICOM reading approaches result in the same volumes loaded in Slicer (or that old readers fail but fixed readers succeed).
+    """
+    parent.acknowledgementText = """This work is supported primarily by the National Institutes of Health, National Cancer Institute, Informatics Technology for Cancer Research (ITCR) program, grant Quantitative Image Informatics for Cancer Research (QIICR) (U24 CA180918, PIs Kikinis and Fedorov). We also acknowledge support of the following grants: Neuroimage Analysis Center (NAC) (P41 EB015902, PI Kikinis) and National Center for Image Guided Therapy (NCIGT) (P41 EB015898, PI Tempany).
+    This file was originally developed by Steve Pieper, Isomics, Inc.
+""" # replace with organization, grant and thanks.
+    self.parent = parent
+
+    # Add this test to the SelfTest module's list for discovery when the module
+    # is created.  Since this module may be discovered before SelfTests itself,
+    # create the list if it doesn't already exist.
+    try:
+      slicer.selfTests
+    except AttributeError:
+      slicer.selfTests = {}
+    slicer.selfTests['DICOMReaders'] = self.runTest
+
+  def runTest(self):
+    tester = DICOMReadersTest()
+    tester.runTest()
+
+#
+# qDICOMReadersWidget
+#
+
+class DICOMReadersWidget(ScriptedLoadableModuleWidget):
+
+  def setup(self):
+    # Instantiate and connect widgets ...
+    ScriptedLoadableModuleWidget.setup(self)
+
+    self.layout.addStretch(1)
+
+#
+# DICOMReadersLogic
+#
+
+class DICOMReadersLogic(ScriptedLoadableModuleLogic):
+  """This class should implement all the actual
+  computation done by your module.  The interface
+  should be such that other python code can import
+  this class and make use of the functionality without
+  requiring an instance of the Widget
+  """
+  def __init__(self):
+    ScriptedLoadableModuleLogic.__init__(self)
+
+  def compareNodes(self,volumeNode1,volumeNode2):
```

Comment:
Should this be a bit more extensive? I think it is reasonable to expect a situation where reading of the attributes related to the image geometry can be interpreted differently by different readers (specifically, this can become ambiguous in interpreting MF, or handling slice thickness in single frame objects).